Added checkout.go with associated funciton.

zachary-trudo opened the pull request:
Also added a 'getIndexByTaskID function' will probably move to main.go

Commit:
Added checkout.go with associated funciton. Also added a 'getIndexByTaskID function' will probably move to main.go

Commit:
Updated function for use with Olivier's future changes.

Commit:
Merged my current changes with Olivier's

Commit:
Finishing merge with Olivier

Commit:
Changing how I lock and unlock

kindermoumoute commented on simple-rest-server/main.go:
```diff
@@ -13,6 +13,24 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+func getIndexByTaskID(taskID string) (index int, err error) {
```
taskID should be int type.

kindermoumoute commented on simple-rest-server/checkout.go:
```diff
@@ -0,0 +1,23 @@
+	defer accessTasks.Unlock()
+
+	// Set the task to checked.
+	allTasks[index].checked = true
```
You shouldn't defer the `accessTasks.Unlock()` and move it after this line for readability and preventing bad usage.

Commit:
Finished ups some changes - added remove.

kindermoumoute commented on simple-rest-server/remove.go:
```diff
@@ -0,0 +1,17 @@
+package main
```
This function should be in `delete.go`

kindermoumoute commented on simple-rest-server/post.go:
```diff
@@ -9,6 +9,10 @@ import (
 	"github.com/julienschmidt/httprouter"
 )
 
+type Remove struct {
```
This structure should be in `delete.go`

kindermoumoute commented on simple-rest-server/checkout.go:
```diff
@@ -0,0 +1,22 @@
+	"log"
+)
+
+func checkTaskAsComplete(taskID int) (err error) {
```
This function should be in `put.go`. This is where we should modify a task.

kindermoumoute commented:
The code LGTM. See my comments about where the code should be.

Commit:
Added a homepage

Commit:
Merge branch 'master' into zach

Commit:
Changed locations of functions and removed files.

- No longer need checkout.go, put.go, or remove.go.
- Moved functions to where they should be.
